types/utils.go

```diff
@@ -140,20 +152,21 @@ func GetRandomNextHeader(header Header) Header {
 	return nextHeader
 }
 
-// GetRandomSignedHeader returns a signed header with random data
-func GetRandomSignedHeader() (*SignedHeader, ed25519.PrivKey, error) {
-	height := uint64(rand.Int63()) //nolint:gosec
-	return GetRandomSignedHeaderWith(height, GetRandomBytes(32), nil)
-}
-
 // GetRandomSignedHeaderWith returns a signed header with specified height and data hash, and random data for other fields
-func GetRandomSignedHeaderWith(height uint64, dataHash header.Hash, privKey ed25519.PrivKey) (*SignedHeader, ed25519.PrivKey, error) {
-	valSet := &cmtypes.ValidatorSet{}
-	if privKey != nil {
-		valSet = GetValidatorSet(privKey)
-	} else {
-		valSet, privKey = GetRandomValidatorSetWithPrivKey()
+func GetRandomSignedHeader(heightP *uint64, dataHash header.Hash, privKey ed25519.PrivKey) (*SignedHeader, ed25519.PrivKey, error) {
```

Comment:
What is the reasoning for changing this to a `*uint64`?

Reply:
Also let's update to follow the same pattern. `GetRandomSignedHeader()` and `GetRandomSignedHeaderCustom(options)`

Reply:
I thought that the pattern for `RandomBlock` had big overhead, and since we have uint, there can't be any value which indicates (nil), unlike for hash and privatekey, in case caller wants us to assign a random height. That's why I used a pointer to int, to use nil.
But I'll update it to the same patter you mentioned above 🫡